### PR TITLE
refactor: Make Sealevel RPC client more functional

### PR DESCRIPTION
**Pull request by ameten**
### Description

Make Sealevel RPC client more functional. Move some methods and error mapping into RPC client.

### Backward compatibility

Yes

### Testing

E2E Ethereum + Sealevel test

### CODE REVIEW

**Review by tkporter — COMMENTED** [review]
LGTM, think the delivered logic may need a tweak

File: rust/main/chains/hyperlane-sealevel/src/rpc/client.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+    pub async fn confirm_transaction_with_commitment(
+        &self,
+        signature: &Signature,
+        commitment: CommitmentConfig,
+    ) -> bool {
+        self.0
+            .confirm_transaction_with_commitment(signature, commitment)
+            .await
+            .map_err(|err| warn!("Failed to confirm inbox process transaction: {}", err))
```
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/chains/hyperlane-sealevel/src/rpc/client.rs]: I moved the error handling outside of the method.

File: rust/main/chains/hyperlane-sealevel/src/mailbox.rs [hunk 2]
```diff
@@ -437,25 +428,21 @@ impl Mailbox for SealevelMailbox {
...
-                &processed_message_account_key,
-                CommitmentConfig::finalized(),
-            )
-            .await
-            .map_err(ChainCommunicationError::from_other)?;
+            .get_account_with_finalized_commitment(&processed_message_account_key)
+            .await;
 
-        Ok(account.value.is_some())
+        Ok(if let Err(ContractError(_)) = result {
```
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/chains/hyperlane-sealevel/src/mailbox.rs]: I incorrectly handled an error here. It should have been `Other`.

I split the method into two. It is better this way as you suggested.

**Review by tkporter — APPROVED** [review]
🛳

**Pull request merged**